### ISSUE TITLE
Cannot allocate write+execute. Perhaps Grsec ?

### ISSUE THREAD

**Issue by db306** [nginx-proxy/docker-letsencrypt-nginx-proxy-companion#656]
Hello !

I ran into an issue, I found this comment on the webs but I have no idea whether its even related https://blog.bandhosting.nl/blog/grsec-issue-after-certbot-auto-install

```
Reloading nginx docker-gen (using separate container nginx-gen)...
Reloading nginx (using separate container 87c6b69169eec11efe3b736093d7b5aed79b740ee52220a297f0348ce72fc699)...
Creating/renewal my.app.com certificates... (my.app.com)
2020-05-12 20:13:59,135:INFO:simp_le:1359: Generating new account key
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/simp_le.py", line 1601, in main
    return main_with_exceptions(cli_args)
  File "/usr/lib/python3.8/site-packages/simp_le.py", line 1585, in main_with_exceptions
    persist_new_data(args, existing_data)
  File "/usr/lib/python3.8/site-packages/simp_le.py", line 1443, in persist_new_data
    client = registered_client(
  File "/usr/lib/python3.8/site-packages/simp_le.py", line 1377, in registered_client
    directory = messages.Directory.from_json(net.get(args.server).json())
  File "/usr/lib/python3.8/site-packages/acme/client.py", line 1144, in get
    self._send_request('GET', url, **kwargs), content_type=content_type)
  File "/usr/lib/python3.8/site-packages/acme/client.py", line 1094, in _send_request
    response = self.session.request(method, url, *args, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 670, in urlopen
    httplib_response = self._make_request(
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 381, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 976, in _validate_conn
    conn.connect()
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 342, in connect
    self.ssl_context = create_urllib3_context(
  File "/usr/lib/python3.8/site-packages/urllib3/util/ssl_.py", line 289, in create_urllib3_context
    context.verify_mode = cert_reqs
  File "/usr/lib/python3.8/site-packages/urllib3/contrib/pyopenssl.py", line 438, in verify_mode
    self._ctx.set_verify(_stdlib_to_openssl_verify[value], _verify_callback)
  File "/usr/lib/python3.8/site-packages/OpenSSL/SSL.py", line 1119, in set_verify
    self._verify_helper = _VerifyHelper(callback)
  File "/usr/lib/python3.8/site-packages/OpenSSL/SSL.py", line 337, in __init__
    self.callback = _ffi.callback(
MemoryError: Cannot allocate write+execute memory for ffi.callback(). You might be running on a system that prevents this. For more information, see https://cffi.readthedocs.io/en/latest/using.html#callbacks

Unhandled error has happened, traceback is above
```

Any tips on this would be highly appreciated.

Also I'm running on an Alpine and using https://github.com/evertramos/docker-compose-letsencrypt-nginx-proxy-companion config

**Comment by buchdag** [comment]
Hi.

I have near to zero experience with grsecurity so I won't be of much help with this 😐 

The only issue that I can remember on a similar topic was #228

**Comment by db306** [comment]
This comes out of a fresh install of Alpine, Docker and Docker-compose with very few basic generic libs. The only other thing that comes to my mind is swap for ubuntu or debian.

**Comment by db306** [comment]
For the record, there is a more detailed issue [here](https://github.com/evertramos/docker-compose-letsencrypt-nginx-proxy-companion/issues/205)

**Comment by buchdag** [comment]
The issue actually comes from the ACME client used inside the container, `simp_le`.

It uses `cerbot`'s ACME python module, hence the similar error you found with `cerbot`. Have you tried what is suggested on https://blog.bandhosting.nl/blog/grsec-issue-after-certbot-auto-install ?

Even if the error is probably not related to the fact you're using https://github.com/evertramos/docker-compose-letsencrypt-nginx-proxy-companion, we do not provide specific support for this or other config / forks.

**Comment by buchdag** [comment]
Also take a look at #228 and #344

**Comment by db306** [comment]
I've updated Alpine Linux from 3.7 to 3.11 and this did not solve the issue. I have absolutely no idea how to fix this as I have relatively 0 experience on this matter beyond changing of OS to ubuntu or debian

**Comment by db306** [comment]
I ended up going for a Debian as I could not find a way to fix this issue. However I stumbled onto this note on the docker website and I do not recall doing this. I think I may have not installed these packages whilst docker was working like a charm and did not output any errors. Perhaps someone in the future might run into this issue, perhaps this was what I was missing all along.

```
For alpine, the following dependency packages are needed: py-pip, python-dev, libffi-dev, openssl-dev, gcc, libc-dev, and make.
```
